src/java.desktop/macosx/classes/sun/java2d/opengl/CGLGraphicsConfig.java

```diff
@@ -252,10 +254,28 @@ public SurfaceData createSurfaceData(CFRetainedResource layer) {
     public Image createAcceleratedImage(Component target,
                                         int width, int height)
     {
-        ColorModel model = getColorModel(Transparency.OPAQUE);
-        WritableRaster wr = model.createCompatibleWritableRaster(width, height);
-        return new OffScreenImage(target, model, wr,
+        int transparency = Transparency.OPAQUE;
+        if (target instanceof Window window && !window.isOpaque())
+            transparency = Transparency.TRANSLUCENT;
+        ColorModel model = getColorModel(transparency);
+
+        float scaleFactor = getDevice().getScaleFactor();
+        int fullResWidth = Math.round( scaleFactor * width);
+        int fullResHeight = Math.round( scaleFactor * height);
+        WritableRaster wr = model.createCompatibleWritableRaster(fullResWidth, fullResHeight);
+        OffScreenImage fullResImage = new OffScreenImage(target, model, wr,
                                   model.isAlphaPremultiplied());
+
+        if (scaleFactor == 1)
+            return fullResImage;
+        return new BaseMultiResolutionImage(fullResImage.getScaledInstance(width, height, Image.SCALE_DEFAULT), fullResImage) {
+            @Override
+            public Graphics getGraphics() {
+                Graphics2D returnValue = fullResImage.createGraphics();
+                returnValue.scale(scaleFactor, scaleFactor);
+                return returnValue;
+            }
+        };
     }
```

Comment:
Should this new `createAcceleratedImage(..)` logic be moved to `CGraphicsConfig`?

This looks like copied and pasted code in two subclasses that could be abstracted into the parent class. (I could be wrong here; I just wanted to check opinions...?)

It looks like these two classes (MTLGraphicsConfig and CGLGraphicsConfig) are the only subclasses for CGraphicsConfig.